subsys/dfu/fmfu_fdev/Kconfig

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+menuconfig FMFU_FDEV
+	bool "Full Modem Firmware Upgrade from flash device"
+	depends on CDDL_GEN
+
+
+if (FMFU_FDEV)
+
+choice
+	prompt "FMFU CDDL decoder build strategy"
+	default FMFU_CDDL_DECODER_BUILD_STRATEGY_SOURCE
+	help
+	  The metadata contained in the FMFU binary is encoded using CDDL.
+	  The decoder used for parsing this data is generated using cddl-gen.
+	  The generated decoder is committed to the repository since the format
+	  mostly will remain unchanged. The format is defined in the file
+	  'modem_update.cddl'. New source files must be generated if this file
+	  changes.
+
+config FMFU_CDDL_DECODER_BUILD_STRATEGY_SOURCE
+	bool "Use committed CDDL decoder source files"
+	help
+	  Select this option as long as 'modem_update.cddl' is unchanged.
+
+config FMFU_CDDL_DECODER_BUILD_STRATEGY_GENERATE
+	bool "Use cddl-gen python script to generate CDDL decoder source files"
+	help
+	  Select this option to re-generate the files needed to decode the
+	  modem update CDDL. This is only needed if the format defined in
+	  'modem_update.cddl' has been changed. Run the
+	  'fmfu_cddl_modem_update_install' target to install the generated files
+	  to the source directory.
```

Comment:
Should there be a setting here which allow users to select a custom `cddl` file ?

For the text, something like:
```suggestion
	help
	  Select this option to generate source and header files needed to decode the
	  modem update CDDL. This is only needed when not using the 'modem_update.cddl' 
	  file provided by Nordic Semiconductor.
	  The repository source files can be replaced with the generated decoder files
	  by executing the 'fmfu_cddl_modem_update_install' build target.
```

Reply:
IMO no, no need for selecting a custom cddl. This needs to be in sync with the binaries contained in the official nordic modem firmware releases.

Reply:
maybe highlight this with a warning ?